packages/language-support/src/autocompletion/completionCoreCompletions.ts

```diff
@@ -621,16 +621,24 @@ function completeAliasName({
     dbSchema,
     ExpectedParameterType.String,
   );
-  const rulesCreatingNewAliasOrDb = [
-    CypherParser.RULE_createAlias,
+  const rulesCreatingNewDb = [
     CypherParser.RULE_createDatabase,
     CypherParser.RULE_createCompositeDatabase,
   ];
   // avoid suggesting existing database names when creating a new alias or database
   if (
-    rulesCreatingNewAliasOrDb.some((rule) =>
-      candidateRule.ruleList.includes(rule),
-    )
+    rulesCreatingNewDb.some((rule) => candidateRule.ruleList.includes(rule))
+  ) {
+    return baseSuggestions;
+  }
+
+  // For `CREATE ALIAS $1 FOR DATABASE $2`
+  // Should not suggest $1 but should suggest for $2
+  // Ideally the grammar would separate declaring and referencing database names
+  // relying on the startTokenIndex is brittle as it will change as the grammar changes
```

Comment:
This comment is a little out of date, how about something like this instead
```suggestion
  // For `CREATE ALIAS $1 FOR DATABASE $2`
  // Should not suggest $1 but should suggest for $2
  // so we return base suggestions if we're at the `aliasName` rule 
```